### PR TITLE
solution

### PR DESCRIPTION
**Pull request by YuriyBilskiy**
[DEMO LINK](https://yuriybilskiy.github.io/react_list-of-goods-hooks/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -15,48 +18,93 @@ export const goodsFromServer = [
...
+      preparedGoods.reverse();
+    }
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<string>('');
+  const [reversed, setReversed] = useState<boolean>(false);
```
Comment (etojeDenys) [inline src/App.tsx:57]: it's redundant to specify the primitives
```suggestion
  const [sortField, setSortField] = useState('');
  const [reversed, setReversed] = useState(false);
```

File: src/App.tsx [hunk 2]
```diff
@@ -15,48 +18,93 @@ export const goodsFromServer = [
...
+          className={cn('button', { 'is-warning': reversed })}
+          onClick={() => setReversed(!reversed)}
         >
           Reverse
         </button>
 
         <button
           type="button"
-          className="button is-danger is-light"
+          // className="button is-danger is-light"
```
Comment (etojeDenys) [inline src/App.tsx:92]: remove all comments

File: src/App.tsx [hunk 3]
```diff
@@ -15,48 +18,93 @@ export const goodsFromServer = [
...
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {visibleGoods.map((good, index) => (
+          // eslint-disable-next-line react/no-array-index-key
+          <li key={index} data-cy="Good">
+            {good}
```
Comment (etojeDenys) [inline src/App.tsx:105]: let's just use the `good` as key instead of the index

**Review by etojeDenys — CHANGES_REQUESTED** [review]
try to fix problems with your demo link, or feel free to ask for some help in the chat

File: src/App.tsx [hunk 4]
```diff
@@ -15,48 +18,94 @@ export const goodsFromServer = [
...
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: string,
```
Comment (etojeDenys) [inline src/App.tsx:28]:
```suggestion
  sortField: SortType,
```

**Review by etojeDenys — CHANGES_REQUESTED** [review]
nit:

**Review by xiimiik — APPROVED** [review]
Nice work!

**Pull request closed without merging**